### PR TITLE
Remove native_functions.yaml dependency from Sorting.cpp

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #67875
* #67874
* #67721
* __->__ #66980
* #66979

Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

### CODE REVIEW

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

**Review by dagitses — APPROVED** [review]

File: aten/src/ATen/native/Sorting.cpp [hunk 1]
```diff
@@ -45,6 +45,17 @@ namespace native {
 DEFINE_DISPATCH(sort_stub);
 DEFINE_DISPATCH(topk_stub);
 
+void _fill_indices(const TensorBase &indices, int64_t dim) {
+  auto dim_size = indices.size(dim);
+  auto idx_dim = at::arange(0, dim_size, indices.options().dtype(at::kLong));
+  auto idx_dim_sizes = std::vector<int64_t>(indices.dim(), 1);
+  auto idx_dim_strides = std::vector<int64_t>(indices.dim(), 0);
+  idx_dim_sizes[dim] = dim_size;
```
Comment (dagitses) [inline aten/src/ATen/native/Sorting.cpp:53]: see if this suppresses internal lint warnings:
`assert(dim < indices.dim());`

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

**Review by malfet — APPROVED** [review]

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

**Comment by malfet** [comment]
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

**Comment by malfet** [comment]
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

**Comment by malfet** [comment]
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

**Pull request closed without merging**